include/nifty/graph/rag/get_lifted_edges_from_rag_and_offsets.hxx

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include <random>
+#include <thread>
+#include "xtensor/xexpression.hpp"
+
+#include "xtensor/xview.hpp"
+#include "xtensor/xmath.hpp"
+#include "nifty/graph/undirected_list_graph.hxx"
+
+#include "nifty/array/arithmetic_array.hxx"
+#include "nifty/tools/for_each_coordinate.hxx"
+#include <cstdlib>
+
+#include <pybind11/pybind11.h>
+namespace py = pybind11;
+
+
+namespace nifty{
+    namespace graph{
+
+        template<std::size_t DIM, class RAG>
+        auto computeLiftedEdgesFromRagAndOffsets(
+                const RAG & rag,
+                const std::vector<array::StaticArray<int64_t, DIM>> & offsets,
+                // OUTPUT & outTensor,
+                const int numberOfThreads
+        ) {
+            // Check whether some of the offsets
+            const auto & labels = rag.labels();
+            const auto & shape = rag.shape();
+
+            nifty::parallel::ParallelOptions pOpts(numberOfThreads);
+            nifty::parallel::ThreadPool threadpool(pOpts);
+            const std::size_t actualNumberOfThreads = pOpts.getActualNumThreads();
+            std::vector<std::vector<std::pair<uint64_t,uint64_t>>> longRangePairs(actualNumberOfThreads);
+
+
+            // Look for lifted edges:
+            {
+                py::gil_scoped_release allowThreads;
```

Comment:
If possible this should not done here, but in the python bindings.

Reply:
This is related to the comments I left in #21. 

In this function I am first appending all found lifted edges uvIds to a vector and then at the end I convert the vector to a numpy tensor that it is returned as output. Because of this, I do not know the dimension of this output tensor until I run the function, so I did not find an easy way to define this tensor initially out of the py::gil_scoped_release scope in the python bindings. 

The easiest workaround I found was then to open the gil_scoped_release scope in the function defined in the hxx file.

Reply:
This can be easily changed to avoid including pybind:
just take a reference to a vector of the lr edges with this function and wrap it within a `gil_scoped_release` in the python bindings (the vector needs to be initialized outside the gil release though).
Then, outside of the gil release create the pytensor of the correct size.
Inside of another gil release copy the values from the vector to the pytensor.

Please make these changes, so that the source code can be build without pybind11.

Reply:
Thanks for the suggestion, now I have updated it